### PR TITLE
fix(action): crd resources can be empty

### PR DESCRIPTION
**Pull request by manute**
**What this PR does / why we need it**:

It checks if the resources are empty to see if this issue https://github.com/helm/helm/issues/31552 is fixed
**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains user facing changes (the `docs needed` label should be applied if so)
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by benoittgt** [comment]
Hello @manute 

Thanks for the patch. Could you add a test that reproduces the issue in #31552 to validate the fix and prevent future regressions?

**Comment by manute** [comment]
Done @benoittgt , thanks!

**Review by banjoh — DISMISSED** [review]
Changes look OK

**Review by TerryHowe — COMMENTED** [review]
Since you added the error check, can you add a test for nil File?

**Comment by manute** [comment]
@TerryHowe  It is already there https://github.com/helm/helm/pull/31578/files#diff-b97a6f3723149f75e7945127b215944ea6a752bcf41a2e1c00b65ae0a409bc7cR1081 or do you refer to other nil File test case?

**Comment by TerryHowe** [comment]
> @TerryHowe It is already there https://github.com/helm/helm/pull/31578/files#diff-b97a6f3723149f75e7945127b215944ea6a752bcf41a2e1c00b65ae0a409bc7cR1081 or do you refer to other nil File test case?

Thanks! Sorry I missed that.

**Review by TerryHowe — DISMISSED** [review]
/lgtm

**Comment by robertsirc** [comment]
Hi @manute first thank you for the PR. I pulled it down built it and ran the test and I got the following:

`=== RUN   TestInstall/install_chart_with_only_crds
    helpers_test.go:63: running cmd (attempt 1): install crd-test testdata/testcharts/chart-with-only-crds --namespace default
    helpers_test.go:69: expected no error, got: 'INSTALLATION FAILED: failed to install CRD crds/test-crd.yaml: resources are empty'
--- FAIL: TestInstall/install_chart_with_only_crds (0.00s)`

**Comment by manute** [comment]
@robertsirc fixed! thanks

**Review by TerryHowe — DISMISSED** [review]
/lgtm

**Comment by banjoh** [comment]
@manute do you mind resolving the current conflicts?

**Review by TerryHowe — DISMISSED** [review]
/lgtm

**Review by banjoh — DISMISSED** [review]

**Comment by TerryHowe** [comment]
Broken tests.
```
--- FAIL: TestInstalLCRDs (0.00s)
    install_test.go:1100: 
        	Error Trace:	/home/runner/work/helm/helm/pkg/action/install_test.go:1100
        	Error:      	Received unexpected error:
        	            	failed to install CRD crds/foo.yaml: resources are empty
        	Test:       	TestInstalLCRDs
--- FAIL: TestInstalLCRDs_AlreadyExist (0.00s)
    install_test.go:1155: 
        	Error Trace:	/home/runner/work/helm/helm/pkg/action/install_test.go:1155
        	Error:      	Expected nil, but got: &errors.errorString{s:"failed to install CRD crds/foo.yaml: resources are empty"}
        	Test:       	TestInstalLCRDs_AlreadyExist
FAIL
```

**Comment by manute** [comment]
@TerryHowe fixed, thanks

File: pkg/action/install_test.go [hunk 1]
```diff
@@ -1084,8 +1084,10 @@ func TestInstallSetRegistryClient(t *testing.T) {
 	assert.Equal(t, registryClient, instAction.GetRegistryClient())
 }
 
-func TestInstalLCRDs(t *testing.T) {
-	config := actionConfigFixture(t)
+func TestInstallCRDs(t *testing.T) {
+	config := actionConfigFixtureWithDummyResources(t, createDummyResourceList(false))
+	config.RESTClientGetter = cli.New().RESTClientGetter()
```
Comment (joejulian) [inline pkg/action/install_test.go]: Why are we depending on a real kubeconfig here?

File: pkg/action/install_test.go [hunk 2]
```diff
@@ -1094,13 +1096,21 @@ func TestInstalLCRDs(t *testing.T) {
...
+		assert.Len(t, crdsToInstall, 1)
+		assert.Equal(t, crdsToInstall[0].File.Data, mockFile.Data)
+		require.NoError(t, instAction.installCRDs(crdsToInstall))
+	})
+
+	t.Run("already exist", func(t *testing.T) {
+		assert.Len(t, crdsToInstall, 1)
+		assert.Equal(t, crdsToInstall[0].File.Data, mockFile.Data)
+		require.NoError(t, instAction.installCRDs(crdsToInstall))
+	})
```
Comment (joejulian) [inline pkg/action/install_test.go]: I don't think this works the way you intended. The fake kubeclient doesn't persist created resources or return "already exists" on subsequent creates. The second run will just happily recreate the same resources and pass because you require NoError.

**Review by joejulian — CHANGES_REQUESTED** [review]

File: pkg/action/install_test.go (hunk 1 above)
Comment (manute) [reply pkg/action/install_test.go]: yeah thanks , fixed

File: pkg/action/install_test.go (hunk 2 above)
Comment (manute) [reply pkg/action/install_test.go]: thanks for the feedback, fixed

**Review by joejulian — APPROVED** [review]

**Review by TerryHowe — APPROVED** [review]
/lgtm
